My Home: Update mobile card to align the action style with established pattern

danhauk opened the pull request:
#### Changes proposed in this Pull Request

* Change the button to a borderless button/link on the My Home mobile card to adhere to the pattern we've established with actions at the bottom of other cards on the screen.
* Add an explanation to help users understand they should open the download link on their phone.

#### Screenshots

Before | After
------------ | -------------
<img width="347" alt="image" src="https://user-images.githubusercontent.com/448298/81206793-9a9df480-8f9a-11ea-9ee1-e15479609849.png"> | <img width="327" alt="Screen Shot 2020-05-13 at 16 21 36" src="https://user-images.githubusercontent.com/1233880/81824518-d81ff600-9535-11ea-96f7-78937f3635a3.png">


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Check out this branch or visit calypso.live link
* Visit My Home for any site
* Make sure the the style of the email download link action looks like the screenshot above and is consistent with the other cards.

danhauk committed:
My Home: Update mobile card to align the action style with established pattern

danhauk committed:
My Home: Update mobile card to align the action style with established pattern

danhauk commented:
@Automattic/dotcom-manage-design I'm curious to hear any thoughts on the additional explanation text. It increases the size of the card which I don't like, but just the link doesn't provide context that the email should be opened on their phone.

gwwar commented:
@danhauk I added wp-calypso as a repository to the dotcom-manage-design team, so we can add folks as reviewers.

gwwar commented on client/my-sites/customer-home/cards/features/go-mobile/index.jsx:
```diff
@@ -73,9 +73,14 @@ export const GoMobile = ( { email, sendMobileLoginEmail } ) => {
+					) }
+					<Button className="go-mobile__email-link-button" onClick={ emailLogin } borderless>
+						{ translate( 'Email download link' ) }
+					</Button>
```
🤔 Why not just make this a link?

danhauk replied:
I was using the convention we have (maybe loosely) that "anchors are for navigation, button is meant for interactions" (or something to that effect). It pops up in the linter if I try changing it to an `<a>` tag.

gwwar replied:
If that's the case, we'd probably keep it as a button with button styling, right? Users will expect links to navigate by default.

mmtr replied:
In terms of accessibility, anchors without an `href` should be also avoided: https://accessibility.oit.ncsu.edu/it-accessibility-at-nc-state/developers/accessibility-handbook/mouse-and-keyboard-events/links/link-behavior/

In regards to using a button with a link styling, just noting we already have some instances in My Home, like when skipping a site setup step:

<img width="1064" alt="Screen Shot 2020-05-07 at 09 34 45" src="https://user-images.githubusercontent.com/1233880/81267005-0db07500-9046-11ea-9295-1c2b176e41b2.png">

sixhours replied:
> In regards to using a button with a link styling, just noting we already have some instances in My Home, like when skipping a site setup step

I think those are intended to be a `borderless` button. Could the mobile card button be a borderless button, too, or does that add unnecessary padding around it?

gwwar commented:
For folks reviewing please use a test user account for the email link to work. (It's both an install + one click login).

Overall functionality still works for me. Looks like we need:
* A +1 for design
* String freeze for new text
* Maybe just let the link be link instead of a styled button?

danhauk commented:
@Automattic/editorial Mind a quick pass at the new copy here if you've got the bandwidth?

Aurorum commented on client/my-sites/customer-home/cards/features/go-mobile/index.jsx:
```diff
@@ -73,9 +73,14 @@ export const GoMobile = ( { email, sendMobileLoginEmail } ) => {
+					{ translate(
+						'We can email a link to open on your phone to instantly download the app.'
+					) }
+					<Button className="go-mobile__email-link-button" onClick={ emailLogin } borderless>
```
Disclaimer: not tested, but can't the `is-link` class just be added? I _think_ that'd minimise the need for additional styling, and is the usual pattern with these kind of buttons. 

```suggestion
					<Button className="go-mobile__email-link-button is-link" onClick={ emailLogin } borderless>
```

gwwar replied:
Right, I think we do have some styling presets @Aurorum  if we do go down that path.

michelleweber commented:
Too long?

Or let us send you a download link via email, and click it on your phone to get the app.
(Send my download link)

Alt:

Get a download link via email -- click it on your phone to get the app.
(Send my download link)

kwight commented:
> Get a download link via email -- click it on your phone to get the app.

Oooh, I like @michelleweber 's alternate text (that would be an em dash I assume?)

Works great for me. I think it needs a period in the subtitle text too:

<img width="394" alt="Screen Shot 2020-05-07 at 9 39 11 AM" src="https://user-images.githubusercontent.com/349751/81321097-0178e780-9047-11ea-8294-b8daca100b2e.png">

danhauk committed:
Copy updates and style the button as a link

danhauk committed:
Copy updates and style the button as a link

danhauk commented:
Thanks for the help everyone! I kept the button as a button, styled as a link using the class like we already do with the task cards - thanks @mmtr @Aurorum 

Made the suggested copy update and added the period in the subtitle text. I also updated the header to be object-oriented rather than action-oriented, so there's consistency there with the rest of the card headings.

a8ci18n commented on client/my-sites/customer-home/cards/features/go-mobile/index.jsx:
```diff
@@ -44,9 +44,9 @@ export const GoMobile = ( { email, sendMobileLoginEmail } ) => {
 			<div className={ classnames( 'go-mobile__row', { 'has-2-cols': showOnlyOneBadge } ) }>
 				<div className="go-mobile__title">
-					<CardHeading>{ translate( 'Go Mobile' ) }</CardHeading>
+					<CardHeading>{ translate( 'WordPress app' ) }</CardHeading>
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'WordPress Apps' )` *ES Score: 6*

mmtr commented on client/my-sites/customer-home/cards/features/go-mobile/style.scss:
```diff
@@ -6,9 +14,16 @@
 .go-mobile__email-link-button {
-	margin-top: 1rem;
-	width: 100%;
+	cursor: pointer;
```
The pointer cursor is already defined in `.button` so I don't think we need it here:

<img width="255" alt="Screen Shot 2020-05-08 at 10 45 27" src="https://user-images.githubusercontent.com/1233880/81388774-1160fc00-9119-11ea-95cb-7b2f7d2b2753.png">

mmtr replied:
Seems the button is displayed inline, causing a misplacement:

<img width="332" alt="Screen Shot 2020-05-08 at 10 42 41" src="https://user-images.githubusercontent.com/1233880/81388883-3fded700-9119-11ea-86f6-c0763b07a854.png">

Can be solved by setting `display: block;`:

<img width="344" alt="Screen Shot 2020-05-08 at 10 47 35" src="https://user-images.githubusercontent.com/1233880/81388984-64d34a00-9119-11ea-8e4d-9e797af41e3e.png">

michelleweber commented:
And yes, that would be an em-dash; I'm spoiled by WP's automatic transformation of the double dash and assume it will happen everywhere. Sorry for being imprecise!

kwight commented:
Same observation as @mmtr , otherwise ready to roll.

danhauk committed:
Fix display issue and remove unnecessary cursor declaration

danhauk committed:
Fix display issue and remove unnecessary cursor declaration

danhauk commented:
Committed those changes to fix the button display issue. Thanks @kwight and @mmtr 

I'm going to re-run the tests that are timing out before merging.
